test/jdk/java/awt/Scrollbar/AquaLFScrollbarTest/ScrollBarBorderTest.java

```diff
@@ -0,0 +1,190 @@
+                                       + Integer.toHexString(c1));
+                    System.out.println(i + " " + j + " " + "Color2: "
+                                       + Integer.toHexString(c2));
+                    saveImage(image1, "himage1.png");
+                    saveImage(image2, "himage2.png");
+                    throw new RuntimeException("Horizontal border has a thumb in it");
+                }
+            }
+        }
+    }
+
+    public static void testVertical() {
+        // create scroll bar
+        scrollBar = new JScrollBar(JScrollBar.VERTICAL);
+        scrollBar.setSize(HEIGHT, WIDTH);
+        scrollBar.setBorder(new VerticalCustomBorder());
+
+        // paint image with thumb set to 0
+        scrollBar.setValue(0);
+        BufferedImage image1 = new BufferedImage(HEIGHT, WIDTH, TYPE_INT_ARGB);
+        Graphics2D graphics2D = image1.createGraphics();
+        scrollBar.paint(graphics2D);
+        graphics2D.dispose();
+
+        // paint image with thumb set to max value
+        scrollBar.setValue(Integer.MAX_VALUE);
+        BufferedImage image2 = new BufferedImage(HEIGHT, WIDTH, TYPE_INT_ARGB);
+        Graphics2D graphics2D2 = image2.createGraphics();
+        scrollBar.paint(graphics2D2);
+        graphics2D2.dispose();
+
+        // check border for thumb
+        for (int i = WIDTH - BORDER_WIDTH; i < WIDTH; i++) {
+            for (int j = 0; j < HEIGHT; j++) {
+                int c1 = image1.getRGB(j,i);
+                int c2 = image2.getRGB(j,i);
+                if(c1 != c2) {
+                    System.out.println(i + " " + j + " " + "Color1: "
+                                       + Integer.toHexString(c1));
+                    System.out.println(i + " " + j + " " + "Color2: "
+                                       + Integer.toHexString(c2));
+                    saveImage(image1, "vimage1.png");
+                    saveImage(image2, "vimage2.png");
+                    throw new RuntimeException("Vertical border has a thumb in it");
+                }
+            }
+        }
+    }
+
+    private static void saveImage(BufferedImage image, String filename) {
+        try {
+            ImageIO.write(image, "png", new File(filename));
+        } catch (IOException e) {
+            // Don't propagate the exception
+            e.printStackTrace();
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
+        for (UIManager.LookAndFeelInfo laf : UIManager.getInstalledLookAndFeels()) {
+                SwingUtilities.invokeAndWait(() -> setLookAndFeel(laf));
+                SwingUtilities.invokeAndWait(ScrollBarBorderTest::testHorizontal);
+                SwingUtilities.invokeAndWait(ScrollBarBorderTest::testVertical);
+        }
+    }
+
+    // custom border
+    private static class HorizontalCustomBorder implements Border {
+        public void paintBorder(Component c, Graphics g, int x, int y, int width, int height) {
```

Comment:
I suggest adding `@Override` annotation to the methods implementing `Border` interface, that is to all the methods in both Both `HorizontalCustomBorder` and `VerticalCustomBorder` classes.